From 031f6e29be546969cfc56fa1bee6f7718acd3dd8 Mon Sep 17 00:00:00 2001
From: Bill Mills <bill.mills@linaro.org>
Date: Thu, 14 Mar 2024 13:48:32 -0400
Subject: [PATCH] HACK arm64 devicetree Xilinx: match zcu106 DisplayPort for
 kv260

I don't understand what I am doing!
I am just pattern matching here.
I have no idea why port@5 is the correct port number.

zcu106 had this and kv260 did not.
So I copy/pasted these sections to kv260.

I now get HDMI and DP on kv260 on next-20240314
If I use this DTB with v6.6.20 it also works.

In both cases I am using defconfig + my zynqmp-kv260.config [1]

[1] https://github.com/wmamills/yocto-genericarm64/blob/main/kconfig/zynqmp-kv260.config

Signed-off-by: Bill Mills <bill.mills@linaro.org>
---
 .../boot/dts/xilinx/zynqmp-sck-kv-g-revB.dtso | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-sck-kv-g-revB.dtso b/arch/arm64/boot/dts/xilinx/zynqmp-sck-kv-g-revB.dtso
index a7b8fffad499..9ca9c23a7e70 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-sck-kv-g-revB.dtso
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-sck-kv-g-revB.dtso
@@ -52,6 +52,19 @@ si5332_5: si5332-5 { /* u17 */
 		#clock-cells = <0>;
 		clock-frequency = <27000000>;
 	};
+
+	dpcon {
+		compatible = "dp-connector";
+		label = "P11";
+		type = "full-size";
+
+		port {
+			dpcon_in: endpoint {
+				remote-endpoint = <&dpsub_dp_out>;
+			};
+		};
+	};
+
 };
 
 &i2c1 { /* I2C_SCK C23/C24 - MIO from SOM */
@@ -81,6 +94,15 @@ &zynqmp_dpsub {
 	phy-names = "dp-phy0", "dp-phy1";
 	phys = <&psgtr 1 PHY_TYPE_DP 0 0>, <&psgtr 0 PHY_TYPE_DP 1 0>;
 	assigned-clock-rates = <27000000>, <25000000>, <300000000>;
+
+	ports {
+		port@5 {
+			dpsub_dp_out: endpoint {
+				remote-endpoint = <&dpcon_in>;
+			};
+		};
+	};
+
 };
 
 &zynqmp_dpdma {
-- 
2.34.1

